ci/check-protobuf-against-stable.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+readonly SRC_DIR="$(dirname "${SCRIPT_DIR}")"
+
+cd "${SRC_DIR}"
+readonly CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+
+set_latest_stable() {
+  local gitTagSearchSpace
+  if [[ "${CURRENT_BRANCH}" == "main" ]]; then
+    gitTagSearchSpace="--no-merged"
+  else
+    gitTagSearchSpace="--merged"
+  fi
+  LATEST_STABLE="$(git tag ${gitTagSearchSpace} | grep -v "snapshot" | sort -V | tail -1)"
+}
+
+cleanup_tmp_files() {
+  rm -rf "${TMP_STABLE_PROTOS_DIR}"
+}
+
+checkout_stable_protos() {
+  trap cleanup_tmp_files EXIT
+  TMP_STABLE_PROTOS_DIR="$(mktemp -d -t ci-stable-protos-XXXXXXXXXX)"
+  local archive;
+  archive="${TMP_STABLE_PROTOS_DIR}/archive.tar.gz"
+  git archive --output="${archive}" "${LATEST_STABLE}" daml-lf ledger-api ledger
+  tar xzf "${archive}" -C "${TMP_STABLE_PROTOS_DIR}"
```

Comment:
This seems a bit complicated at first glance. How is it better than
```
BUF_IMAGE="$(mktemp -d)/buf.tmp"
git checkout $LATEST_STABLE
buf build -o "$BUF_IMAGE"
git checkout main
buf breaking --against "$BUF_IMAGE"
```
?

Reply:
I tried not to touch the main work tree as I don't know much how it is used. I'll do as you suggest.

Reply:
This resets all its changes so shouldn't be an issue, especially with `BUF_IMAGE` out of the worktree. Even if it crashes, the worst case is it leaves the worktree with the wrong commit checked out. Every single job starts by checking out the commit it wants, so this will not affect other jobs, and this is the last step in this job so shouldn't be an issue on that front either.

Reply:
Actually, the current code is easier to test locally and (at least locally when using `direnv`) `dade assist` is not invoked again.

Reply:
Done, ed1ba01